pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -20,25 +20,46 @@ import (
 	"context"
 	"fmt"
 
+	"go.uber.org/zap"
+
 	"github.com/knative/pkg/logging"
 	"github.com/knative/pkg/logging/logkey"
 	kpav1alpha1 "github.com/knative/serving/pkg/apis/autoscaling/v1alpha1"
+	"github.com/knative/serving/pkg/apis/serving"
 	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
 	"github.com/knative/serving/pkg/reconciler/revision/resources"
 	resourcenames "github.com/knative/serving/pkg/reconciler/revision/resources/names"
-	"go.uber.org/zap"
+
+	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/equality"
 	apierrs "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
 )
 
+func (c *Reconciler) resolveDeployment(rev *v1alpha1.Revision) (*appsv1.Deployment, error) {
+	deployments, err := c.deploymentLister.Deployments(rev.Namespace).List(
+		labels.SelectorFromSet(map[string]string{
+			serving.RevisionLabelKey: rev.Name,
+			serving.RevisionUID:      string(rev.UID),
+		}),
+	)
+	if err != nil {
+		return nil, err
+	}
+	if len(deployments) == 0 {
+		return nil, apierrs.NewNotFound(corev1.Resource("Deployments"), rev.Name)
+	}
+	return deployments[0], nil
```

Comment:
Handle the cases where we have 2?

Reply:
Probably here we wouldn't since we don't generate the name, but use constant name.